File: modules/agent/agents/oracle.go
package agents

import (
	"github.com/woyin/OrangeCoding/modules/ai"
	"github.com/woyin/OrangeCoding/modules/core"
	"github.com/woyin/OrangeCoding/modules/tools"
)

// NewOracle creates the question-answering agent.
// RoleObserver, tools: [read_file, grep].
func NewOracle(provider ai.AiProvider, registry *tools.ToolRegistry, workDir string) *BaseAgent {
	allowedTools := []string{"read_file", "grep"}
	return newFilteredAgent(provider, registry, workDir, core.RoleObserver, allowedTools,
		"You are Oracle, the answerer. You answer questions accurately by searching and reading relevant files. You provide clear, well-structured explanations.")
}

File: packages/agent/src/agents/oracle.ts
/**
 * Oracle - question-answering agent.
 * Ported from modules/agent/agents/oracle.go.
 */

import type { AiProvider } from "@orangecoding/ai";
import { AgentRole } from "@orangecoding/core";
import { ToolRegistry } from "@orangecoding/tools";
import { BaseAgent } from "./base.js";
import { newFilteredAgent } from "./helpers.js";

export function newOracle(provider: AiProvider, registry: ToolRegistry, workDir: string): BaseAgent {
  return newFilteredAgent(provider, registry, workDir, AgentRole.Observer,
    ["read_file", "grep"],
    "You are Oracle, the answerer. You answer questions accurately by searching and reading relevant files. You provide clear, well-structured explanations.");
}
